docker/docker_client.go

```diff
@@ -292,10 +313,12 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 		registry = dockerV1Hostname
 	}
 
-	client, err := newDockerClientWithDetails(sys, registry, username, password, "", nil, "")
+	client, err := newDockerClient(sys, registry, registry)
```

Comment:
One more step to the fix: the `reference` argument should be the original value of `registry`, before it gets mapped to `dockerV1Hostname`.

Reply:
Done